service/src/main/java/com/fecklessweasel/service/UniversityUtil.java

```diff
@@ -22,22 +22,36 @@
      * Private constructor to prevent instantiation.
      */
     private static String uniName;
-    private UniversityUtil() { }
+    private static int univid;
+    private UniversityUtil() {
+    }
+
     public static University findUniversity(HttpServletRequest request)
             throws ServiceException {
-
-        // AWAITING CODE FROM ANJANA's branch.
-        /*   //create university by looking up reques
-        uniName = request.getParameter("id");
+        //create university by looking up request
+        univid = Integer.parseInt(request.getParameter("uid"));
```

Comment:
You *must* null check request.getParameter() and try/catch Integer.parseInt() or page will explode if we receive no uid or a uid that is not an integer.